test/specs/behaviors/testHelper.tsx

```diff
@@ -60,25 +67,39 @@ export class TestHelper {
   public findRegexAndAssingCorrespondingInfoToArray(behaviorMenuItems: any) {
     behaviorMenuItems.forEach(behavior => {
       behavior.variations.forEach(variant => {
-        variant.text.split('\n').forEach(singleLineText => {
-          this.iterateRegexDefinitions(singleLineText, variant.name)
-        })
+        if (!variant.specification && !variant.text) {
+          this.createMissingDescriptionTest(variant.name)
+        }
+        if (!variant.specification) {
+          this.verifySpecificationTag(variant.name)
+        } else {
+          variant.specification.split('\n').forEach(singleLineText => {
+            this.iterateRegexDefinitions(singleLineText, variant.name)
+          })
+        }
       })
     })
   }
 
   public iterateRegexDefinitions(singleLineText: string, behaviorName: string) {
+    let regMatched = false
     this.testDefinitions.forEach(testDefinition => {
       const regex = new RegExp(testDefinition.regexp)
       const result = regex.exec(singleLineText)
       if (result) {
+        regMatched = true
         this.filteredDescriptionWithAssignedTestMethod.push({
```

Comment:
this should now be renamed to something like `filteredSpecificationWithAssignedTestMethod` - this also attributes to related types like `FilteredDescription -> FilteredSpecification`. Otherwise the logic will be read in misleading way